### PR TITLE
1301 use wails for front end

### PR DESCRIPTION
**Pull request by thomas-senechal**
Please do not waste your time reviewing:
- `_test.go` files: Mainly changes due to linter update. Tests were not modified.
- Files in `web/massastation/wailsjs/`: Those are generated files from Wails.

## Delivery Quality Checklist

- [ ] **Breaking Changes in API:**
  Does this PR introduce breaking changes in the API?
    - If yes, have you considered making it backward compatible?
    - If backward compatibility is not considered, set the "breaking-change" label.

- [ ] **Changelog:**
  - [ ] For bugfix PR, set the "bugfix" label
  - [ ] If this change should not appear in changelog, use "ignore-for-changelog" label

- [ ] **Version Update Handling:**
  Have you ensured that the version update by user is handled correctly?

- [ ] **PR Dependency:**
  Does this PR depend on another PR?
    - If yes, is it necessary for the dependency to be released prior to merging this one?

- [ ] **Documentation:**
  - [ ] Are any necessary changes made to user-facing documentation?
  - [ ] Confirm that API documentation is updated with any relevant changes.
  - [ ] Check that README and other documentation files are accurate and current.

### CODE REVIEW

**Comment by thomas-senechal** [comment]
[massastation_linux_amd64_bin.zip](https://github.com/massalabs/station/files/13986320/massastation_linux_amd64_bin.zip)

**Comment by thomas-senechal** [comment]
Current state: 
- MacOS is working fine following some discussions with Wails dev and PRs sent to Wails.
- Windows, there are still a few bugs on Windows that need to be fixed such as a crash when right clicking the systray and a Window not appearing. We have some fixes that will be fixed with PRs to be send to Wails repo.
- Linux, extremely unstable. It looks like the Linux implementation lacks of event handling at the moment. Which means that we can't modify the systray right and left click behaviors. We can't reopen an hidden Window due to this lack of event (event not triggered -> callback that handles hide instead of close never called). I tried to add event handling but I couldn't do it in a few hours due to a lack of knowledge on GTK.

In conclusion, Wails v3 will be a great product once ready. But for now, the alpha versions lacks of features and are still too unstable for MS, even for an alpha release. We might come back to Wails for MS once it is more stable

So, **Moving MS to Wails is in standby** and this issue will be closed.

**Comment by thomas-senechal** [comment]
Linux feedback on Wails discord: https://discord.com/channels/1042734330029547630/1199715431997517925/1199715431997517925
